From 7ccac9860ee3b6bd763e1ba15f52a947a9c03edc Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Fri, 25 Feb 2022 10:39:13 +0100
Subject: [PATCH 008/841] feat(stm32mp1): generate default metadata with bl2
 image

When BL2 is build with the PSA_FWU_SUPPORT, generate the
default metadata required to boot the firmware.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git e45676c2a63c6030673584606dc0c46e723380ef feat(stm32mp1): generate default metadata with bl2 image]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: Id216d38ab22025a725608bd222afb2a914038050
---
 plat/st/stm32mp1/platform.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plat/st/stm32mp1/platform.mk b/plat/st/stm32mp1/platform.mk
index ddc528925..f697293f7 100644
--- a/plat/st/stm32mp1/platform.mk
+++ b/plat/st/stm32mp1/platform.mk
@@ -96,6 +96,10 @@ ifeq (${PSA_FWU_SUPPORT},1)
 # Number of banks of updatable firmware
 NR_OF_FW_BANKS			:=	2
 NR_OF_IMAGES_IN_FW_BANK		:=	1
+JSON_METADATA			?=	plat/st/stm32mp1/default_metadata.json
+STM32_DEPS			+=	${BUILD_PLAT}/metadata.bin
+
+$(eval $(call GEN_METADATA,${JSON_METADATA},${BUILD_PLAT}/metadata.bin))
 
 FWU_MAX_PART = $(shell echo $$(($(STM32_TF_A_COPIES) + 2 + $(NR_OF_FW_BANKS))))
 ifeq ($(shell test $(FWU_MAX_PART) -gt $(PLAT_PARTITION_MAX_ENTRIES); echo $$?),0)
-- 
2.47.2

